src/pal/src/misc/miscpalapi.cpp

```diff
@@ -50,6 +50,80 @@ SET_DEFAULT_DEBUG_CHANNEL(MISC);
 static const char RANDOM_DEVICE_NAME[] ="/dev/random";
 static const char URANDOM_DEVICE_NAME[]="/dev/urandom";
 
+
+/*++
+
+Initialization logic for LTTng tracepoint providers.
+
+--*/
+#if defined(__LINUX__)
+
+static const char tpLibName[] = "libcoreclrtraceptprovider.so";
+
+
+/*++
+
+NOTE: PAL_InitializeTracing MUST NOT depend on anything in the PAL itself
+as it is called prior to PAL initialization.
+
+--*/
+static
+void
+PAL_InitializeTracing(void)
```

Comment:
The call to PAL_InitializeTracing is auto-generated by the compiler and called as part of module load (similar to DLLMain on Windows).

If you look at the pal.h, you'll see it's marked as __attribute__((constructor)), which tells the compiler to treat it as a "module constructor".